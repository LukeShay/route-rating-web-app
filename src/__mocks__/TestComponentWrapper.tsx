/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IViewContextState } from "../context/view/viewStore";
import { IUserContextState } from "../context/user/userStore";
import { StaticRouter } from "react-router";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperProps {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  location: string;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapper: React.FC<ITestComponentWrapperProps> = ({
  children,
  gymsState,
  location,
  userState,
  viewState
}): JSX.Element => {
  return (
    <StaticRouter location={`https://lukeshay.com/${location}`}>
      <GymsStoreMock state={gymsState} dispatch={() => {}}>
        <ViewStoreMock state={viewState} dispatch={() => {}}>
          <UserStoreMock state={userState} dispatch={() => {}}>
            {children}
          </UserStoreMock>
        </ViewStoreMock>
      </GymsStoreMock>
    </StaticRouter>
  );
};

export default TestComponentWrapper;
