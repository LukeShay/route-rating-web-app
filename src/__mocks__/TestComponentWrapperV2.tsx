/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IUserContextState } from "../context/user/userStore";
import { IViewContextState } from "../context/view/viewStore";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperV2Props {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapperV2: React.FC<ITestComponentWrapperV2Props> = ({
  children,
  gymsState,
  userState,
  viewState
}): JSX.Element => {
  return (
    <GymsStoreMock state={gymsState} dispatch={() => {}}>
      <ViewStoreMock state={viewState} dispatch={() => {}}>
        <UserStoreMock state={userState} dispatch={() => {}}>
          {children}
        </UserStoreMock>
      </ViewStoreMock>
    </GymsStoreMock>
  );
};

export default TestComponentWrapperV2;
