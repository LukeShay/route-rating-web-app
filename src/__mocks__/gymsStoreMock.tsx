import { GymsContext, IGymsContextState } from '../context/gyms/gymsStore';
import React from 'react';

export interface IGymsStoreMockProps {
  state: IGymsContextState;
  dispatch: any;
}

const GymsStoreMock: React.FC<IGymsStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <GymsContext.Provider value={{ state, dispatch }}>
    {children}
  </GymsContext.Provider>
);

export default GymsStoreMock;
