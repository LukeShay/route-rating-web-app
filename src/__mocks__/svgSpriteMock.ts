export default {
  id: 'svg-sprite-id',
  viewBox: '0 0 1 1',
  url: '/sprite.svg#svg-sprite-id',
};
