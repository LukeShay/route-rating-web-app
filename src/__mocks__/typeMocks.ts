import { Gym, GymPageType, Route, User, Wall, WallTypes } from '../types';
import { GymsContextState } from '../context/gyms/gymsStore';
import { ViewContextState } from '../context/view/viewStore';
import { UserContextState } from '../context/user/userStore';

export const testRouteOne: Route = {
  averageGrade: 'GRADE_5_9',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Blue',
  id: 'route1',
  name: 'RouteOne',
  setter: 'LukeOne',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testRouteTwo: Route = {
  averageGrade: 'GRADE_5_12ab',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Red',
  id: 'route2',
  name: 'RouteTwo',
  setter: 'LukeTwo',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testWallOne: Wall = {
  gymId: 'gym1',
  id: 'wall1',
  name: 'Wall1',
  routes: [testRouteOne, testRouteTwo],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testWallTwo: Wall = {
  gymId: 'gym1',
  id: 'wall2',
  name: 'Wall2',
  routes: [],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testGymOne: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym1',
  logoUrl: '',
  name: 'Gym One',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testGymTwo: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym2',
  logoUrl: '',
  name: 'Gym Two',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testUserEditor: User = {
  authority: 'ADMIN',
  city: 'Ames',
  country: 'USA',
  email: 'email',
  firstName: 'Name',
  id: 'id',
  lastName: 'Last',
  password: 'password',
  phoneNumber: '1111111111',
  role: 'ADMIN_ROLE',
  session: null,
  state: 'IA',
  username: 'username',
};

export const mockGymsState: GymsContextState = {
  page: { content: [testGymOne, testGymTwo] } as GymPageType,
};

export const darkDesktopState: ViewContextState = {
  mobile: false,
  theme: 'DARK_THEME',
};

export const signedInEditorState: UserContextState = {
  user: testUserEditor,
};

export const signedOutState: UserContextState = {
  user: null,
};
