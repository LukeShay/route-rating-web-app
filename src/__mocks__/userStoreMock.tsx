import { IUserContextState, UserContext } from '../context/user/userStore';
import React from 'react';

export interface IUserStoreMockProps {
  state: IUserContextState;
  dispatch: any;
}

const UserStoreMock: React.FC<IUserStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <UserContext.Provider value={{ state, dispatch }}>
    {children}
  </UserContext.Provider>
);

export default UserStoreMock;
