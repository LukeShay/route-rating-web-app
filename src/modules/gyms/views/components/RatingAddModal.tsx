import { Button } from '@material-ui/core';
import GradeIcon from '@material-ui/icons/Grade';
import React from 'react';
import { toast } from 'react-toastify';
import * as GymsActions from '../../../../context/gyms/gymsActions';
import { useGymsContext } from '../../../../context/gyms/gymsStore';
import { Gym, RouteRating } from '../../../../types';
import Form from '../../../common/forms/Form';
import Input from '../../../common/inputs/Input';
import Selector from '../../../common/inputs/Select';
import TransitionModal from '../../../common/modal/Modal';

export interface IRatingAddModalProps {
  gym: Gym;
  open: boolean;
  routeId: string;
  wallId: string;
  handleClose(): Promise<void> | void;
}

const RatingAddModal: React.FunctionComponent<IRatingAddModalProps> = ({
  gym,
  open,
  routeId,
  wallId,
  handleClose,
}): JSX.Element => {
  const [rating, setRating] = React.useState<number>(0);
  const [grade, setGrade] = React.useState<string>('');
  const [review, setReview] = React.useState<string>('');
  const [reviewMessage, setReviewMessage] = React.useState<string>(
    '0/140 characters used.'
  );

  const { dispatch: gymsDispatch } = useGymsContext();

  const handleChange = async (event: any): Promise<void> => {
    const { id, value } = event.target;

    if (id === 'rating') {
      setRating(value);
    } else if (id === 'grade') {
      setGrade(value);
    } else if (id === 'review' && value.length <= 140) {
      setReviewMessage(value.length + '/140 characters used.');
      setReview(value);
    }
  };

  const handleSubmit = (event: any): void => {
    event.preventDefault();

    if (rating !== 0 && grade !== '') {
      GymsActions.createRouteRating(
        gymsDispatch,
        {
          grade,
          rating,
          review,
          routeId,
        } as RouteRating,
        gym,
        wallId
      ).then((response: Response) => {
        if (!(response instanceof Response) || !response.ok) {
          toast.error('Error creating review.');
        } else {
          handleClose();
        }
      });
    }
  };

  const FormHead: JSX.Element = (
    <div style={{ display: 'inline' }}>
      <div style={{ float: 'left', marginRight: '25px', marginTop: '5px' }}>
        Add rating
      </div>
      <div style={{ float: 'right', marginLeft: '25px' }}>
        <Button onClick={handleClose} type="button" variant="outlined">
          Cancel
        </Button>
      </div>
    </div>
  );

  const FormBody: JSX.Element = (
    <React.Fragment>
      <Selector
        handleChange={handleChange}
        value={rating}
        id="rating"
        name="rating-select"
        label="Rating"
        native={true}
      >
        <option value="" />
        <option value={1}>1</option>
        <option value={2}>2</option>
        <option value={3}>3</option>
        <option value={4}>4</option>
        <option value={5}>5</option>
      </Selector>
      <Selector
        handleChange={handleChange}
        value={grade}
        id="grade"
        name="grade-select"
        label="Grade"
        native={true}
      >
        <option value="" />
        <option value={'GRADE_5_5'}>5.5</option>
        <option value={'GRADE_5_6'}>5.6</option>
        <option value={'GRADE_5_7'}>5.7</option>
        <option value={'GRADE_5_8'}>5.8</option>
        <option value={'GRADE_5_9'}>5.9</option>
        <option value={'GRADE_5_10ab'}>5.10ab</option>
        <option value={'GRADE_5_10cd'}>5.10cd</option>
        <option value={'GRADE_5_11ab'}>5.11ab</option>
        <option value={'GRADE_5_11cd'}>5.11cd</option>
        <option value={'GRADE_5_12ab'}>5.12ab</option>
        <option value={'GRADE_5_12cd'}>5.12cd</option>
        <option value={'GRADE_5_13ab'}>5.13ab</option>
        <option value={'GRADE_5_13cd'}>5.13cd</option>
        <option value={'GRADE_5_14ab'}>5.14ab</option>
        <option value={'GRADE_5_14cd'}>5.14cd</option>
        <option value={'GRADE_5_15ab'}>5.15ab</option>
        <option value={'GRADE_5_15cd'}>5.15cd</option>
      </Selector>
      <Input
        autoCapitalize={'true'}
        fullWidth={true}
        value={review}
        onChange={handleChange}
        type="text"
        id="review"
        rows={4}
        helpText={reviewMessage}
        error={review.length === 140}
      />
    </React.Fragment>
  );
  return (
    <TransitionModal
      id="ratingAdd"
      open={open}
      handleClose={handleClose}
      style={{ width: '475px' }}
    >
      <Form
        id="ratingAdd"
        title={FormHead}
        formInputs={FormBody}
        buttonText="Save"
        handleSubmit={handleSubmit}
        icon={
          <div style={{ display: 'inline', paddingBottom: '10px' }}>
            <GradeIcon fontSize="small" />
            <GradeIcon fontSize="small" />
            <GradeIcon fontSize="small" />
            <GradeIcon fontSize="small" />
            <GradeIcon fontSize="small" />
          </div>
        }
      />
    </TransitionModal>
  );
};

export default RatingAddModal;
