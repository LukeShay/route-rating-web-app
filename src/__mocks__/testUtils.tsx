import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import TestComponentWrapperV2 from './TestComponentWrapperV2';
import * as TypeMocks from './typeMocks';

interface WrapperProps {
  children: React.ReactNode;
}

const SignedInDesktop: React.FC<WrapperProps> = ({ children }): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedInEditorState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedInDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedInDesktop, ...options });

const SignedOutDesktop: React.FC<WrapperProps> = ({
  children,
}): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedOutState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedOutDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedOutDesktop, ...options });

export * from '@testing-library/react';
