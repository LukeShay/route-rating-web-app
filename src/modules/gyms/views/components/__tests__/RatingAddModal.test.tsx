import React from "react";
import RatingAddModal from "../RatingAddModal";
import { shallow } from "../../../../../../configs/setupEnzyme";
import * as TypeMocks from "../../../../../__mocks__/typeMocks";
import { shallowToJson } from "enzyme-to-json";

let view;

describe("<RatingAddModal />", () => {
  beforeEach(() => {
    view = shallow(
      <RatingAddModal
        gym={TypeMocks.testGymOne}
        open={true}
        routeId={TypeMocks.testRouteOne.id}
        wallId={TypeMocks.testWallOne.id}
        handleClose={() => {}}
      />
    );
  });

  it("should render correctly", () => {
    expect(shallowToJson(view)).toMatchSnapshot();
  });

  it("should have correct fields", () => {
    expect(view.find("select[id='rating']")).toBeDefined();
    expect(view.find("select[id='grade']")).toBeDefined();
    expect(view.find("textarea[id='review']")).toBeDefined();
    expect(view.find("button[data-test-id='submit-button-test-id']")).toBeDefined();
  });
});
