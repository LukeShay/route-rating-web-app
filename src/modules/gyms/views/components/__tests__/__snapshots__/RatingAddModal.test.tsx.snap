// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingAddModal /> should render correctly 1`] = `
<TransitionModal
  handleClose={[Function]}
  id="ratingAdd"
  open={true}
  style={
    Object {
      "width": "475px",
    }
  }
>
  <Form
    buttonText="Save"
    formInputs={
      <React.Fragment>
        <Selector
          handleChange={[Function]}
          id="rating"
          label="Rating"
          name="rating-select"
          native={true}
          value={0}
        >
          <option
            value=""
          />
          <option
            value={1}
          >
            1
          </option>
          <option
            value={2}
          >
            2
          </option>
          <option
            value={3}
          >
            3
          </option>
          <option
            value={4}
          >
            4
          </option>
          <option
            value={5}
          >
            5
          </option>
        </Selector>
        <Selector
          handleChange={[Function]}
          id="grade"
          label="Grade"
          name="grade-select"
          native={true}
          value=""
        >
          <option
            value=""
          />
          <option
            value="GRADE_5_5"
          >
            5.5
          </option>
          <option
            value="GRADE_5_6"
          >
            5.6
          </option>
          <option
            value="GRADE_5_7"
          >
            5.7
          </option>
          <option
            value="GRADE_5_8"
          >
            5.8
          </option>
          <option
            value="GRADE_5_9"
          >
            5.9
          </option>
          <option
            value="GRADE_5_10ab"
          >
            5.10ab
          </option>
          <option
            value="GRADE_5_10cd"
          >
            5.10cd
          </option>
          <option
            value="GRADE_5_11ab"
          >
            5.11ab
          </option>
          <option
            value="GRADE_5_11cd"
          >
            5.11cd
          </option>
          <option
            value="GRADE_5_12ab"
          >
            5.12ab
          </option>
          <option
            value="GRADE_5_12cd"
          >
            5.12cd
          </option>
          <option
            value="GRADE_5_13ab"
          >
            5.13ab
          </option>
          <option
            value="GRADE_5_13cd"
          >
            5.13cd
          </option>
          <option
            value="GRADE_5_14ab"
          >
            5.14ab
          </option>
          <option
            value="GRADE_5_14cd"
          >
            5.14cd
          </option>
          <option
            value="GRADE_5_15ab"
          >
            5.15ab
          </option>
          <option
            value="GRADE_5_15cd"
          >
            5.15cd
          </option>
        </Selector>
        <Input
          autoCapitalize="true"
          error={false}
          fullWidth={true}
          helpText="0/140 characters used."
          id="review"
          name=""
          onChange={[Function]}
          placeholder=""
          rows={4}
          type="text"
          value=""
        />
      </React.Fragment>
    }
    handleSubmit={[Function]}
    icon={
      <div
        style={
          Object {
            "display": "inline",
            "paddingBottom": "10px",
          }
        }
      >
        <Memo(GradeIcon)
          fontSize="small"
        />
        <Memo(GradeIcon)
          fontSize="small"
        />
        <Memo(GradeIcon)
          fontSize="small"
        />
        <Memo(GradeIcon)
          fontSize="small"
        />
        <Memo(GradeIcon)
          fontSize="small"
        />
      </div>
    }
    id="ratingAdd"
    title={
      <div
        style={
          Object {
            "display": "inline",
          }
        }
      >
        <div
          style={
            Object {
              "float": "left",
              "marginRight": "25px",
              "marginTop": "5px",
            }
          }
        >
          Add rating
        </div>
        <div
          style={
            Object {
              "float": "right",
              "marginLeft": "25px",
            }
          }
        >
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
            type="button"
            variant="outlined"
          >
            Cancel
          </WithStyles(ForwardRef(Button))>
        </div>
      </div>
    }
  />
</TransitionModal>
`;
